.changeset/eighty-nights-vanish.md

```diff
@@ -0,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+Add `experimental.preserveOriginalMain` option to `unstable_readConfig`. This will pass the original `main` value through, without converting it to an absolute path.
```

Comment:
Do we really need to call it experimental option considering the API is still unstable?

I think `experimental` usually refers to experimental flag on wrangler.

Reply:
Good point. I'll remove the experimental naming.

Reply:
https://github.com/cloudflare/workers-sdk/pull/10212/commits/be0d1c467d9d8f943c679ae51eba73084cc9fac1